code/datums/mood.dm

```diff
@@ -0,0 +1,471 @@
+	if(mob_to_make_moody.hud_used)
+		modify_hud()
+		var/datum/hud/hud = mob_to_make_moody.hud_used
+		hud.show_hud(hud.hud_version)
+
+/datum/mood/Destroy(force, ...)
+	STOP_PROCESSING(SSmood, src)
+
+	QDEL_LIST_ASSOC_VAL(mood_events)
+
+	unmodify_hud()
+	var/mob/living/mob_parent = parent?.resolve()
+	if (mob_parent)
+		mob_parent.lose_area_sensitivity(MOOD_DATUM_TRAIT)
+		UnregisterSignal(mob_parent, list(COMSIG_MOB_HUD_CREATED, COMSIG_ENTER_AREA, COMSIG_LIVING_REVIVE))
+
+	return ..()
+
+/datum/mood/process(delta_time)
+	var/mob/living/mob_parent = parent?.resolve()
+	if (mob_parent.stat == DEAD)
+		return
+
+	switch(mood_level)
+		if(1)
+			set_sanity(sanity - 0.3 * delta_time, SANITY_INSANE)
+		if(2)
+			set_sanity(sanity - 0.15 * delta_time, SANITY_INSANE)
+		if(3)
+			set_sanity(sanity - 0.1 * delta_time, SANITY_CRAZY)
+		if(4)
+			set_sanity(sanity - 0.05 * delta_time, SANITY_UNSTABLE)
+		if(5)
+			set_sanity(sanity, SANITY_UNSTABLE) //This makes sure that mood gets increased should you be below the minimum.
+		if(6)
+			set_sanity(sanity + 0.2 * delta_time, SANITY_UNSTABLE)
+		if(7)
+			set_sanity(sanity + 0.3 * delta_time, SANITY_UNSTABLE)
+		if(8)
+			set_sanity(sanity + 0.4 * delta_time, SANITY_NEUTRAL, SANITY_MAXIMUM)
+		if(9)
+			set_sanity(sanity + 0.6 * delta_time, SANITY_NEUTRAL, SANITY_MAXIMUM)
+	handle_nutrition()
+
+	// 0.416% is 15 successes / 3600 seconds. Calculated with 2 minute
+	// mood runtime, so 50% average uptime across the hour.
+	if(HAS_TRAIT(mob_parent, TRAIT_DEPRESSION) && DT_PROB(0.416, delta_time))
+		add_mood_event("depression_mild", /datum/mood_event/depression_mild)
+
+	if(HAS_TRAIT(mob_parent, TRAIT_JOLLY) && DT_PROB(0.416, delta_time))
+		add_mood_event("jolly", /datum/mood_event/jolly)
+
+/// Handles mood given by nutrition
+/datum/mood/proc/handle_nutrition()
+	var/mob/living/mob_parent = parent?.resolve()
+	if (HAS_TRAIT(mob_parent, TRAIT_NOHUNGER))
+		return FALSE // no moods for nutrition
+	switch(mob_parent.nutrition)
+		if(NUTRITION_LEVEL_FULL to INFINITY)
+			if (!HAS_TRAIT(mob_parent, TRAIT_VORACIOUS))
+				add_mood_event(MOOD_CATEGORY_NUTRITION, /datum/mood_event/fat)
+			else
+				add_mood_event(MOOD_CATEGORY_NUTRITION, /datum/mood_event/wellfed) // round and full
+		if(NUTRITION_LEVEL_WELL_FED to NUTRITION_LEVEL_FULL)
+			add_mood_event(MOOD_CATEGORY_NUTRITION, /datum/mood_event/wellfed)
+		if( NUTRITION_LEVEL_FED to NUTRITION_LEVEL_WELL_FED)
+			add_mood_event(MOOD_CATEGORY_NUTRITION, /datum/mood_event/fed)
+		if(NUTRITION_LEVEL_HUNGRY to NUTRITION_LEVEL_FED)
+			clear_mood_event(MOOD_CATEGORY_NUTRITION)
+		if(NUTRITION_LEVEL_STARVING to NUTRITION_LEVEL_HUNGRY)
+			add_mood_event(MOOD_CATEGORY_NUTRITION, /datum/mood_event/hungry)
+		if(0 to NUTRITION_LEVEL_STARVING)
+			add_mood_event(MOOD_CATEGORY_NUTRITION, /datum/mood_event/starving)
+
+/**
+ * Adds a mood event to the mob
+ *
+ * Arguments:
+ * * category - (text) category of the mood event
```

Comment:
could you explain in more detail what the category is

Reply:
https://github.com/timothymtorres/tgstation/blob/2d0e0826d34bb1cc1f8b6070b926b444892910e9/code/datums/mood_events/_mood_event.dm#L16-L19

Also most mood events don't even need category.  It should be an optional argument.